maths/montecarlo.py

```diff
@@ -0,0 +1,43 @@
+"""
+@author: MatteoRaso
+"""
+
+
+def piEstimator(iterations: int):
+    """An implementation of the Monte Carlo method used to find pi.
+    1. Draw a 2x2 square centred at (0,0).
+    2. Inscribe a circle within the square.
+    3. For each iteration, place a dot anywhere in the square.
+    3.1 Record the number of dots within the circle.
+    4. After all the dots are placed, divide the dots in the circle by the total.
+    5. Multiply this value by 4 to get your estimate of pi.
+    6. Print the estimated and numpy value of pi
+    """
+    from numpy import pi, sqrt
+    from random import uniform
+
+    circle_dots = 0
+
+    # A local function to see if a dot lands in the circle.
+    def circle(x: float, y: float):
```

Comment:
Why nesting function, if not using them as decorators or closure?

Reply:
It is legit to only allow the nested function to be called from it’s parent function.

Reply:
But what does it achieves, maybe define the nested function on the same level as parent function, or maybe using a lambda(at least in this case).

Reply:
It is only designed to be called inside the main function.  I find a nested function to be more natural and understandable than a lambda.